### ISSUE TITLE
Error: unknown base "N" in `matrixOfRandomBarcodes`

### ISSUE THREAD

**Issue by j-andrews7** [crisprVerse/screenCounter#3]
I run into this error with certain samples, and not others. Nothing obvious wrong with the FASTQs. I initially though this might be due to either reads with all Ns or starting with an N, as they haven't been trimmed or filtered for quality, but I was unable to replicate it with the example code throwing barcodes/sequences of all Ns at it.

```
fastqs <- my_file_paths

barcodes <- c()
tab <- data.frame()
for (i in seq_along(fastqs)) {
  name <- fqs[[i]]
  message(paste0("Counting ", name))
  out <- matrixOfRandomBarcodes(fastqs[[i]],
      template="NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCAGAG", num.threads = 8)
  
  barcodes <- c(barcodes, rowData(out)$sequences)
  
  # Summary stats
  stats <- cbind("Sample" = name, 
               "Total Reads" = out$nreads, 
               "Total Reads with Barcode" = out$nmapped, 
               "Total Barcodes" = nrow(out), 
               "Barcodes Found Once" = length(which(assay(out) == 1)), 
               "% Barcodes with 1 Read" = 100 * (length(which(assay(out) == 1)) / nrow(out)), 
               "% Barcodes with 2-5 Reads" = 100 * (length(which(assay(out) > 1 & assay(out) < 6)) / nrow(out)), 
               "% Barcodes with >5 Reads" = 100 * (length(which(assay(out) > 5)) / nrow(out)))
  
  tab <- rbind(tab, stats)
}
```

Result:
```
Error: BiocParallel errors
  1 remote errors, element index: 1
  0 unevaluated and other errors
  first remote error:
Error: unknown base 'N'
```

**Comment by LTLA** [comment]
Looks like there are `N`'s in the random barcode; this is causing some degree of unhappiness when the function tries to take the reverse complement. This should be fixed in 1.1.3.

**Comment by j-andrews7** [comment]
Does appear to be fixed, thanks.
